core/common/boot/src/main/java/org/eclipse/edc/boot/system/runtime/BaseRuntime.java

```diff
@@ -135,6 +142,27 @@ protected ServiceExtensionContext createServiceExtensionContext(Config config) {
         return context;
     }
 
+    /**
+     * Sets the ConsoleMonitor level from config, if specified. The config has precedence over the program argument.
+     */
+    @NotNull
+    protected String[] setLogLevelProgArgFromConfig(Config config, Monitor monitor, String[] programArgs) {
+        if (monitor instanceof ConsoleMonitor) {
+            var levelConfig = config.getString(ConsoleMonitor.LOG_LEVEL_CONFIG, null);
+            if (levelConfig != null) {
+                var logLevelArgs = String.format("%s=%s", ConsoleMonitor.LEVEL_PROG_ARG, levelConfig);
+                var newProgArgs = new ArrayList<>(Arrays.asList(programArgs));
+                newProgArgs = (ArrayList<String>) newProgArgs.stream()
+                                .filter(arg -> !arg.startsWith(ConsoleMonitor.LEVEL_PROG_ARG))
+                                .collect(Collectors.toList());
```

Comment:
this could be simplified by `Stream.of(programArgs)`.
Putting the `LEVEL_PROG_ARG` into the `ConsoleMonitor` class is incorrect, b/c loglevels aren't restricted to the ConsoleMonitor.

Reply:
Other Monitor implementations might have a specific way to set the log level.
The LEVEL_PROG_ARG is here because it's specific to the connector default monitor implementation.